### PR TITLE
Add support for exponents on floats

### PR DESCRIPTION
**Pull request by andimiller**
This allows you to have exponents on double literals.

### CODE REVIEW

File: test/src/test/resources/io.idml.ast/LiteralSuite.json [hunk 1]
```diff
@@ -68,5 +68,12 @@
     "mapping": "negativezero = -0.0",
     "input": {},
     "output": {"negativezero": 0.0}
+  },
+
+  {
+    "name": "floats with exponents",
+    "mapping": "number = 1.23E2",
+    "input": {},
+    "output": {"number": 1.23E2 }
```
Comment (jcalonso) [inline test/src/test/resources/io.idml.ast/LiteralSuite.json]: Could you add another test to test for lowecase `e` and negative sign to have full coverage of the new Antlr grammar?

Comment (andimiller) [reply test/src/test/resources/io.idml.ast/LiteralSuite.json]: yeah I can, I'll chuck them in master, since I've already started the release
